### PR TITLE
Make FreeTrialSpec use GPAlloc

### PR DESCRIPTION
**Pull request by ghost**
**Note to reviewer**: you may want to ignore whitespace

`FreeTrialSpec - A user whose status is Enabled should be able to see the free trial banner, enroll and get terminated` now uses GPAlloc along with the `adoptProject` and `scratchProject` endpoints on orch.

This addresses [firecloud-app#30](https://app.zenhub.com/workspace/o/databiosphere/firecloud-app/issues/30) and [firecloud-app#92](https://app.zenhub.com/workspace/o/databiosphere/firecloud-app/issues/92)

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: If you changed a URL that is used elsewhere (e.g. in an email), comment about where it is used and ensure the dependent code is updated.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [ ] **Submitter**: If you're adding new automated UI tests, review the test plan with QA
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Submitter**: Verify all tests go green, including CI tests and automated UI tests.
- [ ] **Submitter**: Squash commits and merge to develop. If adding test code, merge application code and test code at the same time.
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/user/FreeTrialSpec.scala [hunk 1]
```diff
@@ -64,63 +59,65 @@ class FreeTrialSpec extends FreeSpec with BeforeAndAfterEach with Matchers with
...
-        val billingAccountUponTermination = Google.billing.getBillingProjectAccount(billingProject.get)(trialAuthToken)
-        val errMsg = "The trial user's billing project should have been removed from the billing account."
-        assert(billingAccountUponTermination.isEmpty, errMsg)
+          val billingAccountUponTermination = Google.billing.getBillingProjectAccount(billingProject.get)(trialAuthToken)
+          val errMsg = "The trial user's billing project should have been removed from the billing account."
+          assert(billingAccountUponTermination.isEmpty, errMsg)
 
-        registerCleanUpForDeleteTrialState()
+          // need to re-add a project owner in order for cleanup to succeed
+          addMembersToBillingProject(project, List(campaignManager.email), BillingProjectRole.Owner)(trialAuthToken)
```
Comment (davidangb) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/user/FreeTrialSpec.scala]: could you add this re-add-project-owner step to a cleanup step? It feels slightly awkward to have here.

**Review by davidangb — COMMENTED** [review]
looks good, with one small request inline.

If this passes tests and we can see that the test is relatively fast, I will be super happy!

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/user/FreeTrialSpec.scala (hunk 1 above)
Comment (ghost) [reply automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/user/FreeTrialSpec.scala]: Is there a way to specify the order of cleanup operations? It's being called after the cleanup code from `withCleanBillingProject`. That code needs an admin user in order to release the project. If I could claim the billing project as the trial billing SA I think we could avoid this, but I don't know how to create `Credentials` for a service account.

**Comment by ghost** [comment]
Failed on `WorkspaceReaderSpec`. `FreeTrialSpec - A user whose status is Enabled should be able to see the free trial banner, enroll and get terminated` passed in 1:12.

**Comment by ghost** [comment]
GPAlloc config updated, FreeTrialSpec shall Pass! jenkins retest

**Comment by ghost** [comment]
dev is back up - jenkins retest

**Comment by ghost** [comment]
DataSpec is passing consistently for me. jenkins retest

**Comment by ghost** [comment]
fiab-start failed - jenkins retest

**Comment by ghost** [comment]
Perhaps this will pass swatomation if no other builds are running... jenkins retest

**Comment by ghost** [comment]
Fiab-start failure - jenkins retest

**Comment by ghost** [comment]
GPAlloc has been updated. I've had this test pass consistently over 10 times on my manual runs. Let's see it pass some PR swatomation tests! jenkins retest

**Comment by ghost** [comment]
115 test failures - many due to timeouts. jenkins retest

**Comment by davidangb** [comment]
7 test failures in https://fc-jenkins.dsp-techops.broadinstitute.org/job/swatomation-pipeline/8615/; investigating ... 
```
[info] *** 7 TESTS FAILED ***
[error] Failed: Total 126, Failed 7, Errors 0, Passed 119, Ignored 4
[error] Failed tests:
[error] 	org.broadinstitute.dsde.firecloud.test.analysis.MethodLaunchSpec
[error] 	org.broadinstitute.dsde.firecloud.test.security.AuthDomainMultiGroupsSpec
[error] 	org.broadinstitute.dsde.firecloud.test.workspace.WorkspaceOwnerSpec
[error] 	org.broadinstitute.dsde.firecloud.test.metadata.DataSpec
[error] 	org.broadinstitute.dsde.firecloud.test.security.AuthDomainGroupSpec
[error] (Test / testOnly) sbt.TestsFailedException: Tests unsuccessful
```

**Comment by davidangb** [comment]
all failed specs passed in manual runs:
MethodLaunchSpec
https://fc-jenkins.dsp-techops.broadinstitute.org/job/swatomation-pipeline/8619/
AuthDomainMultiGroupsSpec
https://fc-jenkins.dsp-techops.broadinstitute.org/job/swatomation-pipeline/8620/
WorkspaceOwnerSpec
https://fc-jenkins.dsp-techops.broadinstitute.org/job/swatomation-pipeline/8621/
DataSpec
https://fc-jenkins.dsp-techops.broadinstitute.org/job/swatomation-pipeline/8623/
AuthDomainGroupSpec
https://fc-jenkins.dsp-techops.broadinstitute.org/job/swatomation-pipeline/8624/

**Comment by davidangb** [comment]
I'm annoyed that everything passes separately, but had lots of failures in a full run. Let's try a full run again: jenkins retest

**Pull request merged**